### PR TITLE
Cleaned up more TODOs.

### PR DESCRIPTION
**Pull request by BigBlueHat**
- Avoid using `.exist`.
- Keep should var around, but fix linting.
- Add VP `id` test.
- Add VP type check.
- Anchor text links to VP section.
- Rename presentation-derive tests to enveloped.
- Update envelope fixtures to use objects.
- Turn on VP enveloped VC tests.
- Move enveloped tests into enveloped section.
- Add failure test of string VC in VP.

### CODE REVIEW

**Review by TallTed — APPROVED** [review]
English looks good. I cannot speak to the code.

File: tests/input/presentation-enveloped-vc-missing-required-type-fail.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "@context": [
+    "https://www.w3.org/ns/credentials/v2"
+  ],
+  "type": [
+    "VerifiablePresentation"
+  ],
+  "verifiableCredential": [{
+    "@context": "https://www.w3.org/ns/credentials/v2",
+    "id": "data:application/vc-ld+jwt;base64url,eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJ2YyI6eyJAY29udGV4dCI6WyJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvdjIiLCJodHRwczovL3d3dy53My5vcmcvbnMvY3JlZGVudGlhbHMvZXhhbXBsZXMvdjIiXSwiaWQiOiJodHRwOi8vZXhhbXBsZS5lZHUvY3JlZGVudGlhbHMvMzczMiIsInR5cGUiOlsiVW5pdmVyc2l0eURlZ3JlZUNyZWRlbnRpYWwiXSwiaXNzdWVyIjoiaHR0cHM6Ly9leGFtcGxlLmVkdS9pc3N1ZXJzLzU2NTA0OSIsInZhbGlkRnJvbSI6IjIwMTAtMDEtMDFUMDA6MDA6MDBaIiwiY3JlZGVudGlhbFN1YmplY3QiOnsiaWQiOiJkaWQ6ZXhhbXBsZTplYmZlYjFmNzEyZWJjNmYxYzI3NmUxMmVjMjEiLCJkZWdyZWUiOnsidHlwZSI6IkJhY2hlbG9yRGVncmVlIiwibmFtZSI6IkJhY2hlbG9yIG9mIFNjaWVuY2UgYW5kIEFydHMifX19LCJpc3MiOiJodHRwczovL2V4YW1wbGUuZWR1L2lzc3VlcnMvNTY1MDQ5IiwianRpIjoiaHR0cDovL2V4YW1wbGUuZWR1L2NyZWRlbnRpYWxzLzM3MzIiLCJzdWIiOiJkaWQ6ZXhhbXBsZTplYmZlYjFmNzEyZWJjNmYxYzI3NmUxMmVjMjEifQo=.0fkQPZOKlD0Sl0A798KBUNMOdGq90McQQIEtKU9tgSd9K1kRcxWVKDXQJcn_FJqLvo2bk793EHk-RTeEL1HyAQ",
+    "type": "EnvelopedVerifiableCredential"
```
Comment (PatStLouis) [inline tests/input/presentation-enveloped-vc-missing-required-type-fail.json:11]:
```suggestion
```
The file title is missing required type, shouldn't the type be removed in this case?

File: tests/10-vcdm2.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import {createRequire} from 'module';
 import {filterByTag} from 'vc-test-suite-implementations';
 import {TestEndpoints} from './TestEndpoints.js';
 
+// eslint-disable-next-line no-unused-vars
```
Comment (aljones15) [inline tests/10-vcdm2.js:14]: This is a chore not suitable to this PR, but we really should figure out where chai.should should live and and an eslint rule for the `.eslintrc.js` file that allows this. I think technically speaking we should init should once before the tests run at all. There's a mocha file for it somewhere.

File: tests/input/presentation-enveloped-vc-missing-required-type-fail.json (hunk 1 above)
Comment (BigBlueHat) [reply tests/input/presentation-enveloped-vc-missing-required-type-fail.json:11]: @PatStLouis check the filename. The missing `VerifiableCredential` type is deliberate to cause a failure scenario: `presentation-enveloped-vc-missing-required-type-fail.json`

**Comment by BigBlueHat** [comment]
@aljones15 I removed the commit that was avoiding `.exist()` use, so this is back to how it was--which was simpler.

**Comment by BigBlueHat** [comment]
@aljones15 @PatStLouis could use a re-review. Rebased and removed those questionable `.exist()` replacements.

File: tests/input/presentation-enveloped-vc-missing-required-type-fail.json (hunk 1 above)
Comment (PatStLouis) [reply tests/input/presentation-enveloped-vc-missing-required-type-fail.json:11]: This is what I'm saying, unless I'm missing the point. The file is name missing required type but the required type is in the vc example? The example is valid when the goal is to make it invalid? There's an EnvelopedVerifiableCredential with the `EnvelopedVerifiableCredential` type?

Comment (BigBlueHat) [reply tests/input/presentation-enveloped-vc-missing-required-type-fail.json:11]: 🤦 You're absolutely right! I had my wires crossed with a different example...

**Review by PatStLouis — APPROVED** [review]
LGTM

**Pull request merged**